File: WoodPeckerDemo/WoodPeckerDemo/Modules/IO/EchoService.m
//
//  EchoService.m
//  WoodPeckerDemo
//
//  Created by xiaogang zhang on 2018/1/17.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

#import "EchoService.h"

@implementation EchoService

+ (void)load {
    [[ADHOrganizer sharedOrganizer] registerService:[EchoService class]];
}

+ (NSString *)serviceName {
    return @"adh.EchoService";
}

+ (NSDictionary<NSString*,NSString *> *)actionList {
    return @{
             @"echo" : NSStringFromSelector(@selector(onRequestEcho:)),
             };
}

- (void)onRequestEcho: (ADHRequest *)request {
    NSDictionary * data = @{
                            @"response" : @"hi, I`m App, What`s up",
                            };
    [request finishWithBody:data];
}

@end

File: WoodPeckerSwiftDemo/WoodPeckerSwiftDemo/Modules/IO/EchoService.swift
//
//  EchoService.swift
//  WoodPeckerSwiftDemo
//
//  Created by xiaogang zhang on 2018/1/21.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

import UIKit

class EchoService: ADHService {

    override class func serviceName() -> String {
        return "adh.EchoService";
    }
    
    override class func actionList() -> ([String:String]) {
        return ["echo": NSStringFromSelector(#selector(EchoService.onRequestEcho(request:)))]
    }
    
    @objc func onRequestEcho(request: ADHRequest) {
        let data = ["response" : "Hi, I`m App, what`s up ?"]
        request.finish(withBody: data)
    }
    
}
